src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs

```diff
@@ -93,13 +93,13 @@ protected override string BuildChallengeUrl(AuthenticationProperties properties,
         }
 
         private void AddQueryString<T>(
-           IDictionary<string, string> queryStrings,
+           IDictionary<string, string?> queryStrings,
            AuthenticationProperties properties,
            string name,
-           Func<T, string> formatter,
+           Func<T, string?> formatter,
```

Comment:
This formatter shouldn't be returning null if the input was non-null.

Reply:
But T could be null.

Reply:
We check for that before calling the formatter, though it looks like we would fall back to passing the default to the formatter which might be null. How about adding a check on line 108 to skip the formatter if defaultValue is null?